clients/retrieval_client.go

```diff
@@ -35,20 +38,32 @@ var _ RetrievalClient = (*retrievalClient)(nil)
 
 func NewRetrievalClient(
 	logger common.Logger,
-	indexedChainState core.IndexedChainState,
+	chainState core.ChainState,
+	indexer indexer.Indexer,
 	assignmentCoordinator core.AssignmentCoordinator,
 	nodeClient NodeClient,
 	encoder core.Encoder,
 	numConnections int,
-) *retrievalClient {
+) (*retrievalClient, error) {
+	indexedState, err := coreindexer.NewIndexedChainState(
+		chainState,
+		indexer,
+	)
+	if err != nil {
+		return nil, err
+	}
+	err = indexedState.Start(context.Background())
```

Comment:
Do we need to `Start` the indexedState from the retrieval client? Currently, it's left as the responsibility of caller to provide an indexedState that's actively indexing. 

Even if we do, I don't think this should be called at initialization. If we need to explicitly start a goroutine, it should be separated out as an explicit method